tests/gitgitgadget.test.ts

```diff
@@ -246,14 +246,26 @@ to have included in git.git [https://github.com/git/git].`);
     mails.splice(0);
     expect(await patches2.generateAndSend(logger, send, undefined,
                                           pullRequestURL))
-        .toEqual("pull.1.v2.git.gitgitgadget@example.com");
+        .toMatch(/pull\.1\.v2\.git\.\d+\.gitgitgadget@example\.com/);
     expect(mails.length).toEqual(5);
     if (await gitCommandExists("range-diff", repo.workDir)) {
         expect(mails[0]).toMatch(/Range-diff vs v1:\n[^]*\n -: .* 4: /);
     }
     expect(await repo.revParse("pr-1/somebody/master-v2")).toBeDefined();
 
-    expect(await notes.get(pullRequestURL)).toEqual({
+    const midCleanupRegex = /git\.\d+\.gitgitgadget@example/g;
+    const midCleanup = "git.gitgitgadget@example";
+
+    const seriesMeta = await notes.get<IPatchSeriesMetadata>(pullRequestURL);
+    if (seriesMeta && seriesMeta.coverLetterMessageId) {
+        seriesMeta.coverLetterMessageId = seriesMeta.coverLetterMessageId
+            .replace(midCleanupRegex, midCleanup);
+    }
+    if (seriesMeta && seriesMeta.referencesMessageIds) {
+        seriesMeta.referencesMessageIds[0] = seriesMeta.referencesMessageIds[0]
+            .replace(midCleanupRegex, midCleanup);
+    }
+    expect(seriesMeta).toEqual({
```

Comment:
One part of this test tries to verify that the Message-ID of the cover letter is properly referenced in the replies. This patch weakens this a bit (it would fail to catch a regression where the `References` misses the timestamp, for example).

But it should be relatively easy to extract the cover letter's Message-ID, and for added brownie points verify that it matches the new format.

Something along these lines:

```typescript
const seriesMeta = await notes.get<IPatchSeriesMetadata>(pullRequestURL);

expect(seriesMeta.coverLetterMessageId).not.toBeNull();
expect(seriesMeta.coverLetterMessageId).toMatch(/pull\.1\.v2\.git\.\d+\.gitgitgadget@example\.com/);

expect(seriesMeta).toEqual({
    [...]
    coverLetterMessageId: seriesMeta.coverLetterMessageId,
    [...]
```

That would build a bit more confidence in me ;-)

Reply:
Likewise, you would need the same trick later on:

```typescript
[...]
Submitted-As: https://dummy.com/?mid=${seriesMeta.coverLetterMessageId}
In-Reply-To: https://dummy.com/?mid=${seriesMeta.coverLetterMessageId}`);
```

Reply:
Changes made and pushed.  May do a later PR to use the baseCommit var in the cat-file test to make it more flexible to changes.